### ISSUE TITLE
Invalid ref error when rendering exported react component from HTML

### ISSUE THREAD

**Issue by c9s** [facebook/react#6667]
I encountered the famous `addComponentAsRefTo` issue when I upgraded react 0.13 to v15.0.1 on the same code.

I've checked the `addComponentAsRefTo Invariant Violation` gist and confirmed that I don't have two copied of react.

The following component code is compiled by webpack with config  `{ external: { 'react': 'React' } }` hence the react library is not bundled in the output js.  The bundle exports react components to window.*, for example `window.Editor = Editor;`

The component is not rendered inside the bundled code, instead, it is rendered from HTML page and bind to an element, like the code below:

```
<!-- .... include react v15, react-dom v15 and react addon v15 in the browser -->
<script>
// runs ReactDOM to render the components ....
$(function() {
  ReactDOM.render(React.createElement(window.Editor, { ..... }), document.getElementById('container'));
});
</script>
```

Here is the component code, which is pretty simple and was simplified to just few lines:

```
var SearchBar = React.createClass({
  getInitialState: function() {
    return {};
  },
  handleChange: function(e) { },
  render: function() {
    return (
      <div className="form-group">
        <input className="form-control"
          type="text"
          onChange={this.handleChange}/>
      </div>
    );
  }
});
export default React.createClass({
  getInitialState: function() {
    return {};
  },
  render: function() {
    return <div className="crud-list-container">
      <SearchBar ref="searchBar"/>
    </div>;
  }
});
```

However, If I put the render method in the bundle code, it works fine. like:

https://jsfiddle.net/dskd5um9/1/

**Comment by c9s** [comment]
OK, I just reproduced my issue in jsfiddle:

https://jsfiddle.net/dskd5um9/2/

**Comment by c9s** [comment]
OK I found the cause.

for any one who might encounter this issue:

It's because I included `react-15.0.1.js` and `react-with-addons-15.0.1.js` in the same time, you can only include one react
